### PR TITLE
feat(sumologicextension): enable Sumo Logic Extension

### LINKED ISSUE #29601: New component: Sumo Logic Extension
### The purpose and use-cases of the new component

Code: https://github.com/SumoLogic/sumologic-otel-collector/tree/main/pkg/extension/sumologicextension

This extension is to be used as part of Sumo Logic collector in conjuction with Sumo Logic Exporter in order to export telemetry data to [Sumo Logic](https://www.sumologic.com/).

It manages:

- authentication (passing the provided credentials to sumologicexporter when configured as extension in the same service)
- registration (storing the registration info locally after successful registration for later use)
- heartbeats

### Example configuration for the component

- `installation_token`: (required) collector installation token for the Sumo Logic service, see
  [help][credentials_help] for more details
- `collector_name`: name that will be used for registration; by default the hostname is used. In the event of a conflict, a timestamp will be appended to the name. See [here][clobber] for more information.
- `collector_description`: collector description that will be used for registration
- `collector_category`: collector category that will be used for registration
- `collector_fields`: a map of key value pairs that will be used as collector
  fields that will be used for registration.
  For more information on this subject please visit [this help document][fields_help]
- `discover_collector_tags`: defines whether to auto-discover collector metadata
  tags (for local services, e.g. mysql) (default: `true`)

  **NOTE**: collector metadata tag auto-discovery is an alpha feature.
- `api_base_url`: base API URL that will be used for creating API requests,
  see [API URLs](#api-urls) details
  (default: `https://open-collectors.sumologic.com`)
- `heartbeat_interval`: interval that will be used for sending heartbeats
  (default: `15s`)
- `collector_credentials_directory`: directory where state files with registration
  info will be stored after successful collector registration
  (default: `$HOME/.sumologic-otel-collector`)
- `clobber`: defines whether to delete any existing collector with the same name. See [here][clobber] for more information.
- `force_registration`: defines whether to force registration every time the
  collector starts.
  This will cause the collector to not look at the locally stored credentials
  and to always reach out to API to register itself. (default: `false`)

  **NOTE**: if clobber is unset (default) then setting this to true will create
  a new collector (with new unique name) on Sumo UI on every collector start
  and create a new one upon registration.
- `ephemeral`: defines whether the collector will be deleted after 12 hours
  of inactivity (default: `false`)
- `time_zone`: defines the time zone of the collector. For a list of all possible
  values, refer to the `TZ database name` or `Time zone abbreviation` columns in
  https://en.wikipedia.org/wiki/List_of_tz_database_time_zones#List
- `backoff`: defines backoff mechanism for retry in case of failed registration.
  [Exponential algorithm](https://pkg.go.dev/github.com/cenkalti/backoff/v4#ExponentialBackOff) is being used.
  - `initial_interval` - initial interval of backoff (default: `500ms`)
  - `max_interval` - maximum interval of backoff (default: `1m`)
  - `max_elapsed_time` - time after which registration fails definitely (default: `15m`)

[credentials_help]: https://help.sumologic.com/docs/manage/security/installation-tokens
[fields_help]: https://help.sumologic.com/docs/manage/fields
[clobber]: https://help.sumologic.com/docs/send-data/installed-collectors/collector-installation-reference/force-collectors-name-clobber/

```
extensions:
  sumologic:
    installation_token: <token>
    collector_name: my_collector
    time_zone: CET

receivers:
  hostmetrics:
    collection_interval: 30s
    scrapers:
      load:

processors:

exporters:
  sumologic:
    auth:
      authenticator: sumologic # Specify the name of the authenticator extension

service:
  extensions: [sumologic]
  pipelines:
    metrics:
      receivers: [hostmetrics]
      processors: []
      exporters: [sumologic]
```

### Telemetry data types supported

- logs
- metrics
- traces

### Is this a vendor-specific component?

- [X] This is a vendor-specific component
- [X] If this is a vendor-specific component, I am proposing to contribute and support it as a representative of the vendor.

### Code Owner(s)

@sumo-drosiek

### Sponsor (optional)

@astencel-sumo 

### Additional context

This extensions works in conjuction with Sumo Logic Exporter, however exporter in this repository is in very old version and is not supported

We decided to go with the extension first, as it doesn't require exporter to work, but exporter in new version requires this extension by default

### PR DESCRIPTION
**Pull request by sumo-drosiek**
**Description:**

Add Sumo Logic Extension to contrib executable. Fixes #29601

**Link to tracking Issue:** #29601

**Testing:** N/A

**Documentation:**

`metadata.yaml`

### CODE REVIEW

**Review by swiatekm — APPROVED** [review]

**Comment by atoulme** [comment]
This doesn't add the extension to the contrib distribution. See https://github.com/open-telemetry/opentelemetry-collector-releases/blob/main/distributions/otelcol-contrib/manifest.yaml for it.
This adds the extension to the otelcontribcol executable used to test lifecycle and other aspects of the extension.

I would recommend you go from development to alpha just to be on the safe side.

**Comment by sumo-drosiek** [comment]
@atoulme It should be better now

**Review by atoulme — APPROVED** [review]

File: .chloggen/drosiek-add-sumologicextension.yaml [hunk 1]
```diff
@@ -0,0 +1,27 @@
+# Use this changelog template to create an entry for release notes.
+
+# One of 'breaking', 'deprecation', 'new_component', 'enhancement', 'bug_fix'
+change_type: new_component
+
+# The name of the component, or a single word describing the area of concern, (e.g. filelogreceiver)
+component: sumologicextension
+
+# A brief description of the change.  Surround your text with quotes ("") if it needs to start with a backtick (`).
+note: enable Sumo Logic Extension
```
Comment (astencel-sumo) [inline .chloggen/drosiek-add-sumologicextension.yaml:10]:
```suggestion
note: promote from `development` to `alpha` stability
```

File: cmd/otelcontribcol/builder-config.yaml [hunk 2]
```diff
@@ -30,6 +30,7 @@ extensions:
   - gomod: github.com/open-telemetry/opentelemetry-collector-contrib/extension/solarwindsapmsettingsextension v0.96.0
   - gomod: github.com/open-telemetry/opentelemetry-collector-contrib/extension/storage/dbstorage v0.96.0
   - gomod: github.com/open-telemetry/opentelemetry-collector-contrib/extension/storage/filestorage v0.96.0
+  - gomod: github.com/open-telemetry/opentelemetry-collector-contrib/extension/sumologicextension v0.95.0
```
Comment (astencel-sumo) [inline cmd/otelcontribcol/builder-config.yaml:33]:
```suggestion
  - gomod: github.com/open-telemetry/opentelemetry-collector-contrib/extension/sumologicextension v0.96.0
```

File: cmd/otelcontribcol/go.mod [hunk 3]
```diff
@@ -82,6 +82,7 @@ require (
 	github.com/open-telemetry/opentelemetry-collector-contrib/extension/solarwindsapmsettingsextension v0.96.0
 	github.com/open-telemetry/opentelemetry-collector-contrib/extension/storage/dbstorage v0.96.0
 	github.com/open-telemetry/opentelemetry-collector-contrib/extension/storage/filestorage v0.96.0
+	github.com/open-telemetry/opentelemetry-collector-contrib/extension/sumologicextension v0.95.0
```
Comment (astencel-sumo) [inline cmd/otelcontribcol/go.mod:85]:
```suggestion
	github.com/open-telemetry/opentelemetry-collector-contrib/extension/sumologicextension v0.96.0
```

**Review by andrzej-stencel — APPROVED** [review]

**Comment by dmitryax** [comment]
The tests are failing

**Pull request merged**